### DISCUSSION TITLE
Replacing iTwin login/authentication process

### DISCUSSION THREAD

**Discussion by ovaisa** [iTwin/itwinjs-core#2012] [Q&A] [answered]
We are developing a service based on iTwin and we have our own user registration and authentication process.  We want our users to not to have to go through a second Bentley iTwin login process since they have already been authenticated by our service.

We have been studying documentation on iTwin authentication, but if there an example code for this, would really appreciate if you can point me to that, or any guidance on architecture/workflow to follow.

Thank you,

Ovais

**Comment by aruniverse** [comment, marked as answer]
Hi,

I would recommend you take look at this example, https://github.com/calebmshafer/itwin-viewer-proxy

Basically we have the viewer point to a proxy to the iTwin Platform to get around the Bentley IDP on the frontend. You should be able to use your own authentication service on the frontend, but will need to use our auth on the backend by using the client credentials workflow.

**Reply by ovaisa** [reply]
Thank you Arun for this very helpful pointer.  We have started looking into it.  Much appreciated!
